File: Bootcamp/Q8_6_hanoi.java
public class Q8_6_hanoi {
    public static void hanoi(int n, char source, char destination,char buffer){
        if(n <= 1){
            System.out.println("Move disk 1 from source "+
            source+" to destination "+destination);
                           return;
        }

        hanoi(n-1,source,buffer,destination);
        System.out.println("Move disk "+ n + " from source " +
        source +" to destination " + destination );
        hanoi(n-1,buffer,destination,source);
    }

    public static void  main(String args[]){
        int n = 4; // Number of disks
        hanoi(n, 'A', 'C', 'B'); // A, B and C are names of rods
    }
}

File: Bootcamp/python/Q8_6_hanoi.py
def hanoi(number,src,dst,helper):
    if number <= 0:
        return
    
    hanoi(number-1,src,helper,dst)
    print(f"move disk {number-1} from {src} to {helper}")
    hanoi(number-1,helper,dst,src)
    print(f"move disk {number-1} from {helper} to {dst}")

hanoi(5,"A","B","C")